### ISSUE TITLE
[A11y] Add debounce to screen reader announcements for SelectPanel

### ISSUE THREAD

**Issue by khiga8** [primer/react#7611]
[Staff only: Slack thread](https://github.slack.com/archives/C01L618AEP9/p1771974449945729)

The built-in announcements for SelectPanel is missing a debounce. It appears to be firing after ever single key stroke resulting in a ton of duplicate messages fired consecutively. Video attached in slack thread -- can re-record when I find time later.

[related code](https://github.com/primer/react/blob/main/packages/react/src/FilteredActionList/useAnnouncements.tsx#L9-L10)

**Comment by mustafajw07** [comment]
@khiga8 / @francinelucca Can i contribute for this feature?

**Comment by francinelucca** [comment]
assigned to you @mustafajw07. Feel free to open a PR!
